lib/matplotlib/widgets.py

```diff
@@ -1002,43 +1002,25 @@ def __init__(self, ax, labels, actives=None):
         if actives is None:
             actives = [False] * len(labels)
 
-        if len(labels) > 1:
-            dy = 1. / (len(labels) + 1)
-            ys = np.linspace(1 - dy, dy, len(labels))
-        else:
-            dy = 0.25
-            ys = [0.5]
-
-        axcolor = ax.get_facecolor()
-
-        self.labels = []
-        self.lines = []
-        self.rectangles = []
-
-        lineparams = {'color': 'k', 'linewidth': 1.25,
-                      'transform': ax.transAxes, 'solid_capstyle': 'butt'}
-        for y, label, active in zip(ys, labels, actives):
-            t = ax.text(0.25, y, label, transform=ax.transAxes,
-                        horizontalalignment='left',
-                        verticalalignment='center')
-
-            w, h = dy / 2, dy / 2
-            x, y = 0.05, y - h / 2
-
-            p = Rectangle(xy=(x, y), width=w, height=h, edgecolor='black',
-                          facecolor=axcolor, transform=ax.transAxes)
+        ys = np.linspace(1, 0, len(labels)+2)[1:-1]
+        text_size = mpl.rcParams["font.size"] / 2
 
-            l1 = Line2D([x, x + w], [y + h, y], **lineparams)
-            l2 = Line2D([x, x + w], [y, y + h], **lineparams)
+        self.labels = [
+            ax.text(0.25, y, label, transform=ax.transAxes,
+                    horizontalalignment="left", verticalalignment="center")
+            for y, label in zip(ys, labels)]
 
-            l1.set_visible(active)
-            l2.set_visible(active)
-            self.labels.append(t)
-            self.rectangles.append(p)
-            self.lines.append((l1, l2))
-            ax.add_patch(p)
-            ax.add_line(l1)
-            ax.add_line(l2)
+        self._squares = ax.scatter(
+            [0.15] * len(ys), ys, marker='s', s=text_size**2,
+            c=["none" for _ in range(len(ys))],
```

Comment:
The reason I used an array instead of just `c="none"` is because when I create the `._rectangles` attribute, `facecolor=self._squares.get_facecolor()` in `Rectangle gives a `ValueError` because `none` is returning an empty array. On the other hand, using `["none"] * len(ys)` returns a zero array as expected. This was unexpected, I'm trying to check why this is happening.

Reply:
The `to_rgba_array` returns a `(0, 4)` shape array for `c="none"` which then gets mapped to the facecolors. This can't be used as a value to `facecolors` though since it requires a non-zero shaped array.

Reply:
Though, since I'm using this information only in the Rectangle patch which will be removed later on, I can work around this and make the scatter function for the squares cleaner with `c="none"`. I'll do that.